### ISSUE TITLE
Bridge names with lower `layer=*` values ​​will block higher layer's bridge edge when displayed.

### ISSUE THREAD

**Issue by LaoshuBaby** [gravitystorm/openstreetmap-carto#4924]
### Expected behavior

<img src="https://github.com/gravitystorm/openstreetmap-carto/assets/42690037/404d2f86-e70a-4efa-b673-a4f0980db38a" width=550px/>

### Actual behavior
The name "環河南路快速道路" of [way51911168](https://www.openstreetmap.org/way/51911168) with `layer=1`
<img src="https://github.com/gravitystorm/openstreetmap-carto/assets/42690037/af51e73a-fea2-41cd-a8ef-aa371d39b8f3" width=550px/>

Shown above [way51919180](https://www.openstreetmap.org/way/51919180) with `layer=3`
<img src="https://github.com/gravitystorm/openstreetmap-carto/assets/42690037/262d4e13-aa1d-4532-a3d3-90d9b6416fcb" width=550px/>

And it blocks the black line on the edge of way51919180 (but according to the layer value, way51919180's layer value is larger and should not be blocked by the rendering of the path from the lower layer)
<img src="https://github.com/gravitystorm/openstreetmap-carto/assets/42690037/79fb55d8-5c06-471f-ab18-53024bddf5f0" width=300px/>

<details>
<summary>
### Screenshots with links illustrating the problem
</summary>

<img src="https://github.com/gravitystorm/openstreetmap-carto/assets/42690037/3a3a1e7e-7649-4a30-ab97-992254301499" width=300px/>
<img src="https://github.com/gravitystorm/openstreetmap-carto/assets/42690037/19ba9daf-8bae-4603-aa6d-95571432d5f2" width=300px/>
</details>

**Comment by pnorman** [comment]
Duplicate of #4401 - what you're pointing is part of the name.
